File: ssh.py
#!/usr/bin/env python3
from argparse import ArgumentParser
import subprocess
import os

ssh_config_file = "~/.ssh/config"

# Returns a list of all hosts
def get_hosts():

    hosts = []

    def parse_file(filename):
        with open(os.path.expanduser(filename)) as f:
            content = f.readlines()

        for line in content:
            line = line.lstrip()
            # Ignore wildcards
            if line.startswith('Host ') and not '*' in line:
                for host in line.split()[1:]:
                    hosts.append(host)
            if line.startswith('Include ') and not '*' in line:
                filename = line.replace('Include ', '').rstrip()
                try:
                    parse_file(filename)
                except OSError as e:
                    print(e)

    parse_file(ssh_config_file)

    # Removes duplicate entries
    hosts = sorted(set(hosts))

    return hosts

# Returns a newline seperated UFT-8 encoded string of all ssh hosts
def parse_hosts(hosts):
    return "\n".join(hosts).encode("UTF-8")

# Executes wofi with the given input string
def show_wofi(hosts):

    command="wofi -p \"SSH hosts: \" -d -i --hide-scroll"
    
    process = subprocess.Popen(command,shell=True,stdin=subprocess.PIPE,stdout=subprocess.PIPE)
    ret = process.communicate(input=hosts)
    host, rest = ret
    return host

# Switches the focus to the given id
def ssh_to_host(host, terminal, ssh_command):
    command = "{terminal} \'{ssh_command} {host}\'".format(terminal=terminal, ssh_command=ssh_command, host=host)
    process = subprocess.Popen(command,shell=True)

# Entry point
if __name__ == "__main__":
    
    parser = ArgumentParser(description="Wofi based ssh launcher")
    parser.add_argument("terminal", help='Terminal command to use')
    parser.add_argument("--ssh-command", dest='ssh_command', default='ssh', help='ssh command to use (default=ssh)')
    args = parser.parse_args()

    hosts = get_hosts()
    parsed_hosts = parse_hosts(hosts)
    
    selected = show_wofi(parsed_hosts)
    
    selected_host = selected.decode('utf-8').rstrip()
    ssh_to_host(selected_host, args.terminal, args.ssh_command)

File: .config/hyprland/ssh.pl
#!/usr/bin/env perl
use strict;
use warnings;
use File::Spec;

my $ssh_config_file = "~/.ssh/config";
my $terminal        = 'alacritty -e zsh -c';
my $ssh_command     = 'ssh';

sub get_hosts {
    my @hosts;

    my $parse_file;
    $parse_file = sub {
        my ($filename) = @_;
        $filename =~ s/^~/$ENV{HOME}/;    # Expand tilde manually

        open my $fh, '<', $filename or do {
            warn "Error opening $filename: $!\n";
            return;
        };

        while ( my $line = <$fh> ) {
            $line =~ s/^\s+//;    # lstrip

            # Ignore wildcards
            if ( $line =~ /^Host\s+/ && $line !~ /\*/ ) {
                my @parts = split /\s+/, $line;
                shift @parts;     # Remove 'Host'
                push @hosts, @parts;
            }
            elsif ( $line =~ /^Include\s+/ && $line !~ /\*/ ) {
                my $include_file = $line;
                $include_file =~ s/^Include\s+//;
                chomp $include_file;
                $parse_file->($include_file);
            }
        }
        close $fh;
    };

    $parse_file->($ssh_config_file);

    # Remove duplicates and sort
    my %seen;
    @hosts = sort grep { !$seen{$_}++ } @hosts;

    return @hosts;
}

sub parse_hosts {
    my (@hosts) = @_;
    return join( "\n", @hosts );
}

sub show_menu {
    my ($hosts) = @_;

    # Check which menu program is available
    my $command;
    if ( system("which wofi > /dev/null 2>&1") == 0 ) {
        $command = 'wofi -p "SSH hosts: " -d -i --hide-scroll';
    }
    elsif ( system("which rofi > /dev/null 2>&1") == 0 ) {
        $command = 'rofi -dmenu -p "SSH hosts: " -i';
    }
    else {
        die "Neither wofi nor rofi is installed\n";
    }

    my $selected = `echo '$hosts' | $command`;
    chomp $selected;
    return $selected;
}

sub ssh_to_host {
    my ( $host, $terminal, $ssh_command ) = @_;
    my $command = sprintf "%s '%s %s'", $terminal, $ssh_command, $host;
    system($command);
}

# Main

my @hosts        = get_hosts();
my $parsed_hosts = parse_hosts(@hosts);

my $selected = show_menu($parsed_hosts);
chomp $selected;

if ($selected) {
    ssh_to_host( $selected, $terminal, $ssh_command );
}